docs/references/t-schnorr-how-it-works.mdx

```diff
@@ -0,0 +1,65 @@
+For this reason, a new derivation scheme is specified here.
+The specified scheme does not make use of _clamping_, because it is unnecessary in the given setting, and satisfies the following requirements:
+
+- Off-chain availability: New public keys can be computed off-chain from a master public key without requiring interaction with the IC.
+- Hierarchical derivation: Derived keys are organized in a tree such that from any public key it is possible to derive new child keys. This tree can have arbitrary depth. The first level is used to derive unique canister-specific keys from the master key.
+- Compatibility: The scheme is compatible with existing standards, and it is simple to implement using existing libraries.
+
+### Conventions
+
+We will assume the elliptic curve cryptography using the field and curve parameters defined by Ed25519 (see [RFC8032, Section 5.1](https://datatracker.ietf.org/doc/html/rfc8032#section-5.1)). Variables below are either:
+
+- Integers modulo the order of the curve (referred to as n).
+- Coordinates of points on the curve.
+- Byte sequences.
+
+Addition (+) of two coordinate pairs is defined as application of the EC group operation.
+Concatenation (||) is the operation of appending one byte sequence onto another.
+
+We assume the following functions:
+
+- point(p): returns the coordinate pair resulting from EC point multiplication (repeated application of the EC group operation) of the Ed25519 base point with the integer p.
+- ser<sub>P</sub>(P): serializes the coordinate pair P = (x,y) as a byte sequence using standard 32-byte compressed form (see [RFC8032, 5.1.2 Encoding](https://datatracker.ietf.org/doc/html/rfc8032#section-5.1.2)).
+- utf8(s): returns the UTF-8 encoding of string s.
+- parse<sub>512</sub>(p): interprets a 64-byte sequence as a 512-bit number, most significant byte first.
+- HKDF(salt,IKM,info,L) -> OKM: HMAC-based key derivation function (see [RFC5869](https://datatracker.ietf.org/doc/html/rfc5869)) using HMAC-SHA512 (see [RFC4231](https://datatracker.ietf.org/doc/html/rfc4231)) calculating L-bytes long output key material (OKM) from (byte sequences) salt, input key material (IKM), and application specific information *info*.
+
+### Extended keys
+
+Public keys are extended with an extra 256 bits of entropy, which extension is called chain code and consists of 32 bytes.
+An extended public key is represented as (K, c), with K = point(k) and c being the chain code, for some private key k.
+Each extended key has 2<sub>32</sub> child keys using indices 0 through 2<sub>32</sub>-1.
```

Comment:
This comment seems BIP32 specific - in our extended case the key has effectively unlimited child keys since any arbitrary bytestring can be used.

Reply:
True! I adapted the sentence to "Each extended key can have an arbitrary number of child keys."